BE/Business.Access.Layer/Services/UserService.cs

```diff
@@ -44,9 +44,24 @@ public async Task<BusinessUserModel> Delete(string mail)
             
         }
 
-        public Task<BusinessUserModel> GetByMail(string mail)
+        public async Task<BusinessUserModel> GetByMail(string email)
         {
-            throw new NotImplementedException();
+            var userForRead = await _unitOfWork.Users.FindSingleAsync(user => user.Email.Equals(email));
+            
+            if(email == null)
+            {
+                throw new KeyNotFoundException("User email has NULL value.");
+            }
+
+            if(userForRead == null)
+            {
+                throw new KeyNotFoundException("User with that email doesn't exist!");
```

Comment:
When we return exception messages we don't want to yell at our users (if the message will be presented to the end-user).
There is no need for exclamation here.

FYI (I've seen that this repeats on someplace): @danielastrbac @srdjanstanojcic @damjan21

Reply:
We just send this message to FE, they can edit it in best way for users. 😄